src/server.ts

```diff
@@ -26,120 +33,170 @@ export class MyAgent extends Agent<Env> {
     apiKey: this.env.GROQ_API_KEY,
   })
   messages = RESET
+  connection!: Connection
 
   constructor(state: DurableObjectState, env: Env) {
     super(state, env)
   }
 
   onConnect(connection: Connection, ctx: ConnectionContext): void | Promise<void> {
+    this.connection = connection
     console.log('Connected to agent')
     // Clear history
     this.messages = RESET
-    this.respond(connection, {
+    this.respond({
       role: 'assistant',
       content: `Hello! I'm your AI assistant. How can I help you today?`,
     })
   }
 
-  respond(connection: Connection, message: CoreAssistantMessage | CoreToolMessage) {
+  respond(message: CoreAssistantMessage | CoreToolMessage) {
     this.messages.push(message)
     try {
       // if (message.role === 'tool') {
       //   console.log(message)
       // } else {
       const { content } = message
       if (typeof content === 'string') {
-        connection.send(JSON.stringify(content))
+        this.connection.send(JSON.stringify(content))
       } else if (Array.isArray(content)) {
         content.forEach((line) => {
           console.log({ line })
           if (line.type === 'text') {
-            if (line.text !== '') connection.send(JSON.stringify(line.text))
+            if (line.text !== '') this.connection.send(JSON.stringify(line.text))
             // } else if (line.type === 'tool-result') {
             //   connection.send(JSON.stringify(line.result))
           } else {
-            connection.send(JSON.stringify(line))
+            this.connection.send(JSON.stringify(line))
           }
         })
       } else {
-        connection.send(JSON.stringify(content))
+        this.connection.send(JSON.stringify(content))
       }
       // }
     } catch (error) {
       console.error('Error processing message:', error)
-      connection.send(`Something went wrong: ${(error as Error).message}`)
+      this.connection.send(`Something went wrong: ${(error as Error).message}`)
     }
   }
 
   async onMessage(connection: Connection, message: WSMessage): Promise<void> {
+    if (message === 'PING') return
+
+    this.connection = connection
     this.messages.push({ role: 'user', content: message as string })
     console.log('Message from client', message)
 
-    await this.#talkToLLM(connection, true)
+    await this.#talkToLLM(true)
   }
 
-  async #talkToLLM(connection: Connection, includeTools: boolean) {
+  async #talkToLLM(includeTools: boolean) {
     try {
       const response = await generateText({
         model: this.groq('llama-3.3-70b-versatile'),
         messages: this.messages,
```

Comment:
The template I had wasn't sending history at all, so I had to build all that. I looked at `agent/examples/playground` but didn't see anything there, so I figured it out myself.

But we should definitely make the template more complete, with things like "reset chat" as well